### ISSUE TITLE
Feature request: Custom IgxIcons should be available across modules.

### ISSUE THREAD

**Issue by luiscla27** [IgniteUI/igniteui-angular#3026]
### Description  
Right now custom IgxIcon's are declared using `this.iconService.addSvgIcon("contains", "/assets/images/svg/contains.svg", "filter-icons");` and they are available to be used only at the module were the `this.iconService.addSvgIcon` function was called.

 * igniteui-angular version: 6.2.0

### Result  
Implementation across multiple modules requires to encapsulate and call the `this.iconService.addSvgIcon` method for every added icon even though they already available (at `igx-svg-container` div-node).

This seems to be related to the fact that current "cachedSvgIcons" variable exists only at the current module "IgxIconService" ([here](https://github.com/IgniteUI/igniteui-angular/blob/40f6a05fee2f57517d6f0717dbbce9db13e5a063/projects/igniteui-angular/src/lib/icon/icon.service.ts#L24)).

### Expected result  
Please one of the following:
1. Cached icons should be stored somewhere else where they would be available to be  used by any other module (something like "window")
2. Update your documentation ([here](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/classes/igxiconservice.html#addsvgicon)) so it includes the note that custom icons will only work at the module were 'addSvgIcon' was called.

**Comment by bkulov** [comment]
@luiscla27 If you provide the `IgxIconService` in the application's `app.module.ts` the same instance of the service will be available in all modules in the application. It's required to register the icons only once and they will be available everywhere. The icons are cached so if you try to register them again you'll get the cached copy.

**Comment by luiscla27** [comment]
Thanks @bkulov!! but sorry i forgot to specify, this issue occurs only at [**lazy-loaded modules**](https://angular.io/guide/router#lazy-loading-route-configuration), you may see an example of the issue here:

https://stackblitz.com/edit/custom-igx-icon

**Comment by bkulov** [comment]
@luiscla27 Thanks for the sample. Will take a look.

**Comment by bkulov** [comment]
@luiscla27 I've created a PR to fix this issue. Angular 6 introduced `providedIn` property of the `@Injectable` decorator and this is the preferred method of making a service available in all modules. I've removed `IgxIconModule.forRoot()` method which we were using before. Now (after we release this fix) in order to use the service you will only have to import the module without providing the service. Providing the service will create a new instance of it.

Thanks for reporting this issue.
